File: SDKs/googleapis.com/datalabeling/v1beta1/python/src/sdk/models/shared/googleclouddatalabelingv1beta1createannotationspecsetrequest.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import googleclouddatalabelingv1beta1annotationspecset as shared_googleclouddatalabelingv1beta1annotationspecset


@dataclass_json
@dataclasses.dataclass
class GoogleCloudDatalabelingV1beta1CreateAnnotationSpecSetRequest:
    r"""GoogleCloudDatalabelingV1beta1CreateAnnotationSpecSetRequest
    Request message for CreateAnnotationSpecSet.
    """
    
    annotation_spec_set: Optional[shared_googleclouddatalabelingv1beta1annotationspecset.GoogleCloudDatalabelingV1beta1AnnotationSpecSet] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('annotationSpecSet') }})
    

File: SDKs/googleapis.com/datalabeling/v1beta1/java/lib/src/main/java/openapisdk/models/shared/GoogleCloudDatalabelingV1beta1CreateAnnotationSpecSetRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GoogleCloudDatalabelingV1beta1CreateAnnotationSpecSetRequest
 * Request message for CreateAnnotationSpecSet.
**/
public class GoogleCloudDatalabelingV1beta1CreateAnnotationSpecSetRequest {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("annotationSpecSet")
    public GoogleCloudDatalabelingV1beta1AnnotationSpecSet annotationSpecSet;
    public GoogleCloudDatalabelingV1beta1CreateAnnotationSpecSetRequest withAnnotationSpecSet(GoogleCloudDatalabelingV1beta1AnnotationSpecSet annotationSpecSet) {
        this.annotationSpecSet = annotationSpecSet;
        return this;
    }
}